Framework/PythonInterface/plugins/algorithms/VesuvioAnalysis.py

```diff
@@ -155,15 +158,24 @@ def validateInputs(self):
         TOF = self.getProperty("TOFRangeVector").value
         if len(TOF) != 3:
             issues["TOFRangeVector"] = "TOFRangeVector should have length 3 (lower, binning, upper)."
-        constraints = self.getProperty("ConstraintsProfileNumbers").value
-        if len(constraints) != 0 and len(constraints) != 2:
-            issues["ConstraintsProfileNumbers"] = "ConstraintsProfileNumbers should either be empty or only contain 2 numbers."
-        #check arithmetic is safe
-        if len(constraints) != 0:
-            cross_section = self.getProperty("ConstraintsProfileScatteringCrossSection").value
-            for ch in cross_section:
-                if ch not in ["+","-","*","/",".","(",")"] and not ch.isdigit():
-                    issues["ConstraintsProfileScatteringCrossSection"]= "Must be a valid mathmatical expression. "+ch
+        constraints: TableWorkspace = self.getProperty("ConstraintsProfile").value
+        if constraints and constraints.rowCount() > 0:
+            if constraints.columnCount()!= len(constraintCols) or \
+                    sorted(cleanNames(constraintCols))!=sorted(cleanNames(constraints.getColumnNames())):
+                issues["ConstraintsProfile"] = "The constraints table should be of the form: "
+                for name in constraintCols:
+                    issues["ConstraintsProfile"] += name + ", "
+            else:
+                #check arithmetic is safe
+                cross_section = constraints.column('ScatteringCrossSection')
+                for section in cross_section:
+                    for ch in section:
+                        if ch not in ["+","-","*","/",".","(",")"] and not ch.isdigit():
```

Comment:
Nit: This can break, since something like `1*/2` would be accepted as valid.

If users run into this as a future bug, we need to be careful not to blindly use `eval` because of the security risks with IDaaS (something like `import os; os.rm("/some/file")` would be possible as a table input). Discussion: https://stackoverflow.com/questions/2371436/evaluating-a-mathematical-expression-in-a-string

But I'll leave this for as/if we ever need to fix this (I suspect it will be fine)

Reply:
Thanks for the info, I will keep that in mind.